Claude/investigate sentry event 01 hi c khct5jtt p bo5 fa dybe6

itnos opened the pull request.

itnos committed:
Update Sentry API URL to use environment variable

itnos committed:
Refactor webhook handling for various event types

itnos committed:
Refactor SentryRequestDataType to use new types

itnos committed:
Refactor webhook handling for better clarity

itnos committed:
Add optional fields for browser, runtime, and URL

itnos committed:
Fix issue with extracting project name from URL

itnos committed:
Add Russian language support for hook messages

itnos committed:
Add support for Russian language in message generation

itnos committed:
Update link text from 'ТУТ' to 'ЗДЕСЬ'

claude committed:
Fix context data extraction for event_alert.triggered events

The data.event handler was only extracting basic fields (title, culprit)
but not the detailed context data (environment, os, device, browser,
runtime, etc.). Also fixed handling of project field which can be
either an object or a number.

Now event_alert.triggered events will show full information in
Telegram messages instead of "none" values.

tuanngocptn commented:
Hey man, you can re-open the PR if you want to merge it in to my Repo. @itnos

itnos commented:
Changes to sentry-telegram-webhook
Repository: https://github.com/itnos/sentry-telegram-webhook (fork of tuanngocptn/sentry-telegram-webhook)
Problem
Original webhook did not work with self-hosted Sentry 25.11:

Hardcoded URL sentry.io
Did not support data.error and data.event structures from Alert Rules
Crashed on failed API requests to Sentry

Modified files
1. src/app.service.ts

Replaced https://sentry.io with process.env.SENTRY_URL

2. src/app.controller.ts

Added support for data.event (Alert Rules) and data.error (webhooks)
Extract project name from event URL
Extract error position with line number from stacktrace
Extract environment, device, OS from contexts
Made getIssueDetail() optional (does not crash on API error)
Added helper methods formatOs() and formatUser()

3. src/app.d.ts

Added types SentryEventType and SentryIssueType
SentryRequestDataType now supports issue?, event?, error?

4. src/app.utils.ts

Changed "App name" to "Project"
Added generateHookMessageRu() function for Russian language

5. src/app.helper.ts

Added support for LANGUAGE=ru

Environment variables
SENTRY_URL=https://example.com  # new variable
LANGUAGE=ru                          # new option

tuanngocptn commented on src/app.controller.ts:
```diff
@@ -35,28 +35,187 @@ export class AppController {
+          appName: projectName,
+          title: title,
+          errorPosition: culprit,
+          detailLink: webUrl || `${process.env.SENTRY_URL}/organizations/${process.env.SENTRY_ORGANIZATION_SLUG}/issues/${issueId}/`,
```
Please add `SENTRY_URL` in to the `.env.template` file.

tuanngocptn commented on src/app.service.ts:
```diff
@@ -34,7 +34,8 @@ export class AppService {
       method: 'get',
-      url: `https://sentry.io/api/0/organizations/${process.env.SENTRY_ORGANIZATION_SLUG}/issues/${issueId}/tags/`,
+      url: `${process.env.SENTRY_URL || 'https://sentry.io'}/api/0/organizations/${process.env.SENTRY_ORGANIZATION_SLUG}/issues/${issueId}/tags/`,
+      // url: `https://sentry.io/api/0/organizations/${process.env.SENTRY_ORGANIZATION_SLUG}/issues/${issueId}/tags/`,
```
Remove the old code please. We don't need it anymore after change to `SENTRY_URL`.

tuanngocptn commented on src/app.controller.ts:
```diff
@@ -35,28 +35,187 @@ export class AppController {
+          issueId = error.issue_id;
+          webUrl = error.web_url || '';
+
+          // Извлекаем имя проекта из issue_url или url
```
please comment with English

tuanngocptn commented on src/app.controller.ts:
```diff
@@ -35,28 +35,187 @@ export class AppController {
+          }
+          
+
+          // Извлекаем данные из contexts
```
please use English too

tuanngocptn commented on src/app.controller.ts:
```diff
@@ -35,28 +35,187 @@ export class AppController {
+          const exception = error.exception?.values?.[0];
+          const frame = exception?.stacktrace?.frames?.slice(-1)[0]; // последний фрейм - место ошибки
+
+          // Позиция с полным путём и номером строки
```
Please using english too 😅

tuanngocptn commented on src/app.controller.ts:
```diff
@@ -35,28 +35,187 @@ export class AppController {
+            culprit = `${filename}:${lineno}${func}`;
+          }
+
+          // Формируем issueDetails из данных error
```
Please using english too 😅

tuanngocptn reviewed (changes requested):
Overal is okay, Please check and change the Readme.md for adding the new ENV that you just added.

claude committed:
Add SENTRY_URL to .env.template

claude committed:
Add SENTRY_URL documentation to README.md

- Added SENTRY_URL to environment variables table
- Added 'ru' to supported languages in the table
- Updated all .env examples with SENTRY_URL
- Updated Docker run command with SENTRY_URL
- Updated all Docker Compose examples with SENTRY_URL

itnos committed:
Clean up getIssueDetail by removing commented URL

Removed commented-out URL in getIssueDetail method.

itnos committed:
Translate comments to English

tuanngocptn commented:
Release [new version 1.1.0](https://github.com/tuanngocptn/sentry-telegram-webhook/releases/tag/1.1.0) for the new change from @itnos

itnos commented:
Hi. Sorry to bother you, but your description hasn't changed here
https://hub.docker.com/r/tuanngocptn/sentry-telegram-webhook

tuanngocptn commented:
> Hi. Sorry to bother you, but your description hasn't changed here https://hub.docker.com/r/tuanngocptn/sentry-telegram-webhook

ah thanks @itnos I just updated it
